Src/IronPython/Runtime/TypecodeOps.cs

```diff
@@ -9,70 +9,87 @@
 using System.Numerics;
 using System.Runtime.InteropServices;
 
+using IronPython.Runtime.Operations;
+using Microsoft.Scripting.Utils;
+
 namespace IronPython.Runtime {
     internal static class TypecodeOps {
+        // TODO: integrate with PythonStruct
 
-        public static bool IsByteFormat([NotNullWhen(true)]string? format) {
-            int flen = format?.Length ?? 0;
-            if (flen == 0 || flen > 2) return false;
-            char fchar = format![flen - 1];
-            return fchar == 'B' || fchar == 'b' || fchar == 'c';
-        }
+        public const string ValidByteorder = "@=<>!";
+        public const string ValidCodes = "xcbB?uhHiIlLqQnNfdspP";
 
-        public static bool TryGetTypecodeWidth([NotNullWhen(true)]string? typecode, out int width) {
-            if (string.IsNullOrEmpty(typecode) || typecode!.Length > 2 || (typecode.Length == 2 && typecode[0] != '@')) {
-                width = 0;
+        public static bool TryDecomposeTypecode(ReadOnlySpan<char> format, out char byteorder, out char code) {
```

Comment:
This is a remnant from when I thought of consolidating `TypecodeOps` with `PythonStruct` code and was making a few provisions to support that. But after reading `_strucs.cs` I think any consolidated solution, though placed in `TypecodeOps` will be largely based on the code from `_strucs.cs` rather than `TypecodeOps.cs`. 

I will remove it in the next cleanup commit, since I see this consolidation work as low priority at this stage.